File: icamera/src/main/java/me/shouheng/icamera/enums/MediaQuality.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.MediaQuality.*;

/**
 * Media quality
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-12-28 16:30
 */
@IntDef(value = {QUALITY_AUTO, QUALITY_LOWEST, QUALITY_LOW, QUALITY_MEDIUM, QUALITY_HIGH, QUALITY_HIGHEST})
@Retention(value = RetentionPolicy.SOURCE)
public @interface MediaQuality {

    /** Auto */
    int QUALITY_AUTO        = 0;

    /** Lowest quality */
    int QUALITY_LOWEST      = 1;

    /** Low quality */
    int QUALITY_LOW         = 2;

    /** Medium quality */
    int QUALITY_MEDIUM      = 3;

    /** High quality */
    int QUALITY_HIGH        = 4;

    /** Highest quality */
    int QUALITY_HIGHEST     = 5;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/MediaQuality.kt
package me.shouheng.icamera.enums

import android.support.annotation.IntDef

/**
 * Media quality
 *
 * @author [WngShhng](mailto:shouheng2015@gmail.com)
 * @version 2019-12-28 16:30
 */
@IntDef(value = [MediaQuality.QUALITY_AUTO,
    MediaQuality.QUALITY_LOWEST,
    MediaQuality.QUALITY_LOW,
    MediaQuality.QUALITY_MEDIUM,
    MediaQuality.QUALITY_HIGH,
    MediaQuality.QUALITY_HIGHEST])
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class MediaQuality {
    companion object {
        /** Auto  */
        const val QUALITY_AUTO = 0
        /** Lowest quality  */
        const val QUALITY_LOWEST = 1
        /** Low quality  */
        const val QUALITY_LOW = 2
        /** Medium quality  */
        const val QUALITY_MEDIUM = 3
        /** High quality  */
        const val QUALITY_HIGH = 4
        /** Highest quality  */
        const val QUALITY_HIGHEST = 5
    }
}